integration-testing/test/conftest.py

```diff
@@ -88,7 +95,7 @@ def temporary_bonds_file(validator_keys):
 
 
 @contextlib.contextmanager
-def validators_data(config):
+def validators_data(config: TestConfig) -> Iterator[ValidatorsData]:
```

Comment:
The return type here should be https://docs.python.org/3/library/typing.html#typing.ContextManager

Reply:
No. If I change it to `ContextManager`, mypy would throw an error, `The return type of a generator function should be "Generator" or one of its supertypes`.

It is a generator func because you use `yield`, so the return type should be `Generator`. The `yield` object should be type inside the `[]`